views/006_config_views.sql

```diff
@@ -654,3 +663,53 @@ CREATE OR REPLACE VIEW config_detail AS
         WHERE start_time > NOW() - interval '30 days' 
         GROUP BY config_id) as playbook_runs
       ON ci.id = playbook_runs.config_id;
+
+-- config parent to config_relationship trigger
+CREATE OR REPLACE FUNCTION insert_parent_to_config_relationship()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF TG_OP = 'INSERT' THEN
+    IF NEW.parent_id IS NOT NULL THEN
+      INSERT INTO config_relationships (config_id, related_id, relation)
+      VALUES (NEW.parent_id, NEW.id, 'hard')
+      ON CONFLICT DO NOTHING;
+    END IF;
+  ELSIF TG_OP = 'UPDATE' THEN
+    IF NEW.parent_id IS DISTINCT FROM OLD.parent_id THEN
+      IF OLD.parent_id IS NOT NULL THEN
+        DELETE FROM config_relationships
+        WHERE config_id = OLD.parent_id AND related_id = NEW.id AND relation = 'hard';
+      END IF;
+
+      IF NEW.parent_id IS NOT NULL THEN
+        INSERT INTO config_relationships (config_id, related_id, relation)
+        VALUES (NEW.parent_id, NEW.id, 'hard')
+        ON CONFLICT DO NOTHING;
+      END IF;
+    END IF;
+  END IF;
+
+  RETURN NULL;
+END
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE TRIGGER fn_insert_parent_to_config_relationship
+AFTER INSERT OR UPDATE ON config_items
+FOR EACH ROW
+EXECUTE PROCEDURE insert_parent_to_config_relationship();
+
+-- Add the hard relationship for records existing before the trigger was added
+INSERT INTO config_relationships (config_id, related_id, relation)
```

Comment:
I think this needs to be moved to an individual file, and not run every time config_views is updated

Reply:
good point 👍